pkgs/development/python-modules/torch/default.nix

```diff
@@ -160,6 +143,12 @@ in buildPythonPackage rec {
     # base is 10.12. Until we upgrade, we can fall back on the older
     # pthread support.
     ./pthreadpool-disable-gcd.diff
+  ] ++ lib.optionals stdenv.isLinux [
+    # Propagate CUPTI to Kineto by overriding the search path with environment variables.
+    (fetchpatch {
+      url = "https://github.com/pytorch/pytorch/pull/108847/commits/7ae4d7c0e2dec358b4fe81538efe9da5eb580ec9.patch";
+      hash = "sha256-skFaDg98xcJqJfzxWk+qhUxPLHDStqvd0mec3PgksIg=";
+    })
```

Comment:
Are `CUPTI_LIBRARY_DIR`/`CUPTI_INCLUDE_DIR` "standard" names somewhere? Is cupti not handled by `FindCUDAToolkit.cmake`?

Reply:
I don't know that they're standardized, but it's used in Kineto. Well, Kineto at least `CUPTI_INCLUDE_DIR` is -- Kineto has its own `CUDA_cupti_LIBRARY` flag as well: https://github.com/pytorch/kineto/blob/6da7e1799e387d3ed3b003f013c05f377227042c/libkineto/CMakeLists.txt#L56.

Samuel opened a patch for upstream PyTorch (https://github.com/pytorch/pytorch/pull/108847) to propagate these to Kineto's CMake.

I haven't had the opportunity to look into whether `FindCUDAToolkit.cmake` uses it, though that would be interesting to see. If it's not, would you mind creating an issue or something similar and pinging me so we can track it?